### ISSUE TITLE
Release KGO OSS v1.2.2

### ISSUE THREAD

**Issue by pmalek** [Kong/kong-operator#109]
## Steps

This release pipeline is under continous improvement. If you encounter any problem, please refer to the [troubleshooting](#troubleshooting) section of this document.
If the troubleshooting section does not contain the answer to the problem you encountered, please create an issue to improve either the pipeline (if the problem is a bug), or this document (if the problem is caused by human error).

- [x] Check the existing [releases][releases] and determine the next version number.
- [x] Check [default versions](#verify-default-hardcoded-versions) of images (see below).
- [x] Check the `CHANGELOG.md` and update it with the new version number. Make sure the log is up to date.
- [x] Ensure that all generators have run properly (e.g. `make generate manifests`) so that updates to things like CRDs are handled for the release, double check that all manifests from `config/samples/` still work as intended.
- [x] Ensure GitHub PAT is still valid (see [GitHub PAT](#github-pat) below).
- [x] From [GitHub release action][release-action], start a new workflow run with the `release` input set to the release tag (e.g. `v0.1.0`).
- [x] Wait for the workflow to complete.
- [x] The CI should create a PR in the [Gateway Operator][kgo-prs] repo that syncs the release branch to the `main` branch. Merge it.
- [x] After the PR is merged, a new release should be created automatically. Check the [releases][releases] page. The release has to be marked manually as `latest` if this is the case.
- [ ] Update the official documentation at [github.com/Kong/docs.konghq.com][docs_repo]
  - [ ] Run post processing script for `docs/api-reference.md`, providing a tagged version of CRD reference from docs repo as an argument, e.g. `app/_src/gateway-operator/reference/custom-resources/1.2.x.md`. This will add the necessary skaffolding so that the reference is rendered correctly on docs.konghq.com.
    Example: `${GATEWAY_OPERATOR_REPO}/scripts/apidocs-gen/post-process-for-konghq.sh ~/docs.konghq.com/app/_src/gateway-operator/reference/custom-resources/1.2.x.md`
  - NOTE: [CLI configuration options docs][cli_ref_docs] should be updated when releasing KGO EE as that's the source of trutgh for those.
    The reason for this is that KGO EE configuration flags are a superset of OSS flags.
- [ ] Proceed to release KGO EE as it relies on OSS releases.

[docs_repo]: https://github.com/Kong/docs.konghq.com/
[cli_ref_docs]: https://docs.konghq.com/gateway-operator/latest/reference/cli-arguments/

## Verify default hardcoded versions

The package [internal/consts][consts-pkg] contains a list of default versions for the operator. These versions should be updated to match the new release. The example consts to look for:

- `DefaultDataPlaneTag`
- `DefaultControlPlaneTag`
- `WebhookCertificateConfigBaseImage`

## GitHub PAT

**Next expiration date**: 2024-10-02

The release workflow uses @team-k8s-bot's GitHub PAT to create a GitHub release and PRs related to it (in [KGO docs][operator-docs],
[OperatorHub][community-operators], etc.). It's named `Kong Gateway operator release pipeline` and is stored in `PAT_GITHUB`
GitHub repository secret to give workflows access to it. It's always generated with 1-year expiration date.

If you find it's expired, make sure to generate a new one and update the `PAT_GITHUB` secret as well as its 1Pass item
`Github team k8s bot - PAT - Kong Gateway operator release token` for redundancy.

[operator-docs]: https://github.com/Kong/gateway-operator-docs
[community-operators]: https://github.com/Kong/k8s-operatorhub-community-operators

### The release needs to be started again with the same tag

If the release workflow needs to be started again with the same input version, the release branch needs to be deleted. The release branch is created by the CI and it's named `release/v<version>`. For example, if the release version is `v0.1.0`, the release branch will be `release/v0.1.0`.

It's only safe to start a release workflow with the version that was previously used if:

- The release PR to the gateway-operator repo is not merged yet.
- The external hub PRs are not merged yet.
- The tag that matches the release version does not exist.

Otherwise, if the above conditions are not meet, the release cannot be restarted. A new release needs to be started with an input version that would be next in semantic versioning.

Steps:

1. Delete the `release/v<version>` branch.
2. Delete the PR created by a release workflow.
3. Update the repository with the correct changes.
4. Start a new release workflow run.

[releases]: https://github.com/Kong/gateway-operator/releases
[release-action]: https://github.com/Kong/gateway-operator/actions/workflows/release.yaml
[consts-pkg]: https://github.com/Kong/gateway-operator/blob/main/internal/consts/consts.go
[operator-hub-community]: https://github.com/k8s-operatorhub/community-operators
[kgo-prs]: https://github.com/Kong/gateway-operator/pulls

**Comment by pmalek** [comment]
Current feature scope proposal:

```

* 71ab67db - (HEAD -> release/1.2.x, origin/release/1.2.x) fix: make creating DataPlanes index conditional on enabling the ControlPlane controller (#103) (#106) (45 minutes ago) <Patryk Małek - patryk.malek@konghq.com>
* 21604489 - fix(dataplane): don't populate replicas when scaling defined (#79) (13 days ago) <Grzegorz Burzyński - czeslavo@gmail.com>
* a3c38929 - chore(deps): bump sigs.k8s.io/kustomize/api from 0.16.0 to 0.17.1 (#77) (2 weeks ago) <dependabot[bot] - 49699333+dependabot[bot]@users.noreply.github.com>
* d6714a2b - chore(deps): bump golang from 1.22.1 to 1.22.2 (#76) (3 weeks ago) <dependabot[bot] - 49699333+dependabot[bot]@users.noreply.github.com>
* 741beece - chore: release template (#74) (3 weeks ago) <Patryk Małek - patryk.malek@konghq.com>
* fecf8295 - fix: fix kustomization ClusterRoleBinding roleRef (#69) (3 weeks ago) <Patryk Małek - patryk.malek@konghq.com>
* 4e516dbb - fix(build): fix debug dockerfile (#71) (3 weeks ago) <Patryk Małek - patryk.malek@konghq.com>
* eb54adeb - docs: update README.md with information about adding CRDs to PROJECT (#70) (4 weeks ago) <Patryk Małek - patryk.malek@konghq.com>
* 081a7ca2 - chore: remove third_party/ dir (#52) (4 weeks ago) <Patryk Małek - patryk.malek@konghq.com>
* 5e5c6f68 - feat(ci): use mise to manage tools dependencies (#17) (4 weeks ago) <Patryk Małek - patryk.malek@konghq.com>
* 0efa16e7 - docs: add CRD reference docs and ensure they are up to date (#48) (4 weeks ago) <Patryk Małek - patryk.malek@konghq.com>
* 4041323c - chore(deps): bump github.com/kong/kubernetes-testing-framework from 0.45.0 to 0.46.0 (#51) (4 weeks ago) <dependabot[bot] - 49699333+dependabot[bot]@users.noreply.github.com>
* 97adac25 - chore: pull_request_template.md added (#50) (4 weeks ago) <Mattia Lavacca - lavacca.mattia@gmail.com>
* 716b6319 - chore: improve README formatting (#49) (4 weeks ago) <Mattia Lavacca - lavacca.mattia@gmail.com>
* 22553c50 - chore: do not report an error when reducing resources receives a NotFound error (#47) (5 weeks ago) <Patryk Małek - patryk.malek@konghq.com>
* 60352968 - chore(deps): bump github.com/golangci/golangci-lint in /third_party (#45) (5 weeks ago) <dependabot[bot] - 49699333+dependabot[bot]@users.noreply.github.com>
* 830abd5a - fix: ensure that Gateway managed DataPlanes are reduced (#43) (5 weeks ago) <Patryk Małek - patryk.malek@konghq.com>
* 31114aaf - tests: fix flaky HTTPRoute creation (#44) (5 weeks ago) <Patryk Małek - patryk.malek@konghq.com>
* e761c0c0 - readme: add a top-level explanation (#42) (5 weeks ago) <Michał Flendrich - michal.flendrich@konghq.com>
* a66e5ab6 - chore: re-add CONTROLLER_DEVELOPMENT_MODE env support (#40) (5 weeks ago) <Patryk Małek - patryk.malek@konghq.com>
* 3e963271 - chore(deps): bump github.com/golangci/golangci-lint in /third_party (#41) (5 weeks ago) <dependabot[bot] - 49699333+dependabot[bot]@users.noreply.github.com>
* f390085c - chore: license report from FOSSA for commit 687fcc5f2944fe70defe24259625bdaa1540d7f8 (#33) (5 weeks ago) <Michał Flendrich - michal.flendrich@konghq.com>
* b0fafa51 - (tag: v1.2.1) chore(release): [bot] 1.2.1 (#39) (5 weeks ago) <Kong's Team k8s bot - team-k8s+github-bot@konghq.com>
```

**Comment by pmalek** [comment]
1.2.2 has been released but its tag needed to be retracted due to what's described in https://github.com/Kong/gateway-operator/pull/206.

Will follow up with 1.2.3
